test/versioned/redis/redis-v4-legacy-mode.test.js

```diff
@@ -16,20 +16,48 @@ const { checkMetrics } = require('./utils')
 // Indicates unique database in Redis. 0-15 supported.
 const DB_INDEX = 2
 
+const pkg = require('redis/package.json')
+const redisVersion = Number(pkg.version.split('.').shift())
+
 test('Redis instrumentation', async function (t) {
   t.beforeEach(async function (ctx) {
     const agent = helper.instrumentMockedAgent()
     const redis = require('redis')
-    const client = redis.createClient({
-      legacyMode: true,
-      port: params.redis_port,
-      host: params.redis_host
-    })
 
-    await client.connect()
-    await client.v4.flushAll()
+    let client
+    if (redisVersion >= 5) {
+      client = await redis.createClient({
+        port: params.redis_port,
+        host: params.redis_host
+      })
+
+      await client.connect()
+      await client.select(DB_INDEX)
+      await client.flushAll()
 
-    await client.v4.select(DB_INDEX)
+      // Upstream claims `.close` is available on the legacy client, but the
+      // actual implementation proves otherwise.
+      const close = client.close.bind(client)
+
+      // We also need the `.select` method that is not carried over.
+      const select = client.select.bind(client)
+
+      client = client.legacy()
+      client.close = close
+      client.select = (idx, callback) => {
+        select(idx).then(() => { callback() }).catch(error => callback(error))
+      }
+    } else {
+      client = redis.createClient({
```

Comment:
I think I wrote this suite and these updates are def valid but the issue we had seen from a customer was also that the v4 interface wasn't working. Now that you refactored this, there are no tests for the v4 interface in legacy mode

Reply:
Here's that PR https://github.com/newrelic/node-newrelic/pull/2200/files

Reply:
we spoke and both agreed that these tests are valid but we will still need tests for the `.v4` functions when running v4. For v5 those are just in redis.test.js so not needed